File: Java/src/Kaprekar_Number.java
/**
 * Kaprekar Number Checker
 * 
 * @author Jaysmito Mukherjee 
 * @version 2019-A-026
 */
import java.util.*;
public class Kaprekar_Number
{
    public static void main()
    {
        System.out.print("\f");
        Scanner sc=new Scanner(System.in);
        System.out.print("Kaprekar Number Checker\n~~~~~~~~ ~~~~~~ ~~~~~~~\nEnter the number: ");
        int n=sc.nextInt();
        int c=0,a=0,b=0,m=n*n,o=n;
        while(o>0)
        {
            c++;
            o=o/10;
        }
        a=m%(int)Math.pow(10,c);
        b=m/(int)Math.pow(10,c);
        if(a+b==n)
            System.out.print(n+" is a Kaprekar number.");
        else
            System.out.print(n+" is not a Kaprekar number.");        

    }
}

File: Python/Kaprekar_Number.py
print("Kaprekar Number Checker\n")
n = int(input("Enter the number : "))
sq = n * n
flag = False
s = 0
for i in range(1, len(str(sq))-1):
	if int(str(sq)[0:i]) + int(str(sq)[i:]) == n:
		flag = True
		break
if flag:
	print(f"{n} is a kaprekar number.")
else :
	print(f"{n} is not a kaprekar number.")